app/org/sagebionetworks/bridge/sms/SmsMessageProvider.java

```diff
@@ -0,0 +1,99 @@
+package org.sagebionetworks.bridge.sms;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sagebionetworks.bridge.BridgeConstants;
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.accounts.Phone;
+import org.sagebionetworks.bridge.models.studies.SmsTemplate;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.google.common.collect.Maps;
+
+public class SmsMessageProvider {
+    private final Study study;
+    private final Map<String,String> tokenMap;
+    private final Phone phone;
+    private final SmsTemplate template;
+    
+    private SmsMessageProvider(Study study, SmsTemplate template, Phone phone, Map<String, String> tokenMap) {
+        this.study = study;
+        this.template = template;
+        this.phone = phone;
+        this.tokenMap = tokenMap;
+    }
+
+    public Study getStudy() {
+        return study;
+    }
+    public SmsTemplate getTemplate() {
+        return template;
+    }
+    public Phone getPhone() {
+        return phone;
+    }
```

Comment:
Should we have a getter for the token map to facilitate testing?

Reply:
I didn't use it so I removed it, but I can add it back. It's consistent with MimeTypeEmailProvider.